src/Svg.ts

```diff
@@ -2045,35 +2050,124 @@ export class SVGDrawing {
         const dotXPosition = dataDotLeftPosition + this.shiftX ;
         const dotYPosition = dataDotTopPosition + this.shiftY ;
 
-        if( dotXPosition >= logoXPosition && 
-            dotXPosition <= logoXPosition + logoXLength && 
-            dotYPosition >= logoYPosition && 
-            dotYPosition <= logoYPosition + logoYLength)
-            return true;
-
-        if( dotXPosition + dotLength >= logoXPosition && 
-            dotXPosition + dotLength <= logoXPosition + logoXLength && 
-            dotYPosition >= logoYPosition && 
-            dotYPosition <= logoYPosition + logoYLength)
-            return true;
-
-        if( dotXPosition >= logoXPosition && 
-            dotXPosition <= logoXPosition + logoXLength && 
-            dotYPosition + dotLength >= logoYPosition && 
-            dotYPosition + dotLength <= logoYPosition + logoYLength)
-            return true;
+        if( dotXPosition >= logoXPosition &&
+            dotXPosition < logoXPosition + logoXLength &&
+            dotYPosition >= logoYPosition &&
+            dotYPosition < logoYPosition + logoYLength) {
+            return true ;
+        }
+        if( dotXPosition + dotLength > logoXPosition &&
+            dotXPosition + dotLength <= logoXPosition + logoXLength &&
+            dotYPosition >= logoYPosition &&
+            dotYPosition < logoYPosition + logoYLength) {
+            return true ;
+        }
 
-        if( dotXPosition + dotLength >= logoXPosition && 
-            dotXPosition + dotLength <= logoXPosition + logoXLength && 
-            dotYPosition + dotLength >= logoYPosition && 
-            dotYPosition + dotLength <= logoYPosition + logoYLength)
-            return true;
+        if( dotXPosition >= logoXPosition &&
+            dotXPosition < logoXPosition + logoXLength &&
+            dotYPosition + dotLength > logoYPosition &&
+            dotYPosition + dotLength <= logoYPosition + logoYLength) {
+            return true ;
+        }
+        if( dotXPosition + dotLength > logoXPosition &&
+            dotXPosition + dotLength <= logoXPosition + logoXLength &&
+            dotYPosition + dotLength > logoYPosition &&
+            dotYPosition + dotLength <= logoYPosition + logoYLength) {
+            return true ;
+        }
 
         return false ;
 
     }   
 
 
+    calculateLogoDimensionsModuleApproach() {
+        let logoHeight , logoWidth , logoAreaWidth , logoAreaHeight;
+
+        let logoScale = this.config.logoScale ;
+        let logoMargin = this.config.logoMargin ;
+        let rectangular = this.config.rectangular ;
+
+        // Calibrating logo margin to avoid small logos
+        // ( This will ensure at least 50% area of logo is covered by the actual logo)
+        if (logoScale <= 0 || logoScale >= 1) {
```

Comment:
Instead of checking ` logoScale >= 1` , can we check ` logoScale >= maxLogoScale` . 
Current implementation will allow values between maxLogoScale and 1 .

Reply:
Done.
Change pushed in develop PR as well